cpp/oneapi/dal/table/backend/interop/host_homogen_table_adapter.cpp

```diff
@@ -350,10 +260,35 @@ host_homogen_table_adapter<Data>::host_homogen_table_adapter(const homogen_table
         return;
     }
 
+    size_t nFeatures = table.get_column_count();
+    size_t nRows = table.get_row_count();
+
+    if (table.get_data_layout() == data_layout::row_major) {
+        base = daal::data_management::HomogenNumericTable<Data>::create(
+            daal::data_management::DictionaryIface::equal,
+            ptr_data_t{ const_cast<Data*>(table.get_data<Data>()), daal_object_owner(table) },
```

Comment:
`const_cast<Data*>(table.get_data<Data>())` create temporary variable to store this. The same for 
`const_cast<Data*>(&(table.get_data<Data>()[i * nRows]))` in the other branch.

Reply:
done